app/scripts/controllers/network/network-controller.test.js

```diff
@@ -50,224 +69,4106 @@ const PRE_1559_BLOCK = {
+
+        const supportsEIP1559 = await controller.getEIP1559Compatibility();
+
+        expect(supportsEIP1559).toBe(false);
+      });
+    });
+
+    it('persists to state that the network does not support EIP-1559 when baseFeePerGas is not in the block header', async () => {
+      await withController(
+        {
+          state: {
+            networkDetails: {
+              EIPS: {},
+            },
+          },
+        },
+        async ({ controller, network }) => {
+          network.mockEssentialRpcCalls({
+            latestBlock: PRE_1559_BLOCK,
           });
-          expect(
-            controller.networkDetails.getState().EIPS[1559],
-          ).toBeUndefined();
+          await controller.initializeProvider();
+
           await controller.getEIP1559Compatibility();
-          expect(controller.networkDetails.getState().EIPS[1559]).toStrictEqual(
+
+          expect(controller.store.getState().networkDetails.EIPS[1559]).toBe(
             false,
           );
+        },
+      );
+    });
+
+    it('does not make multiple requests to eth_getBlockByNumber when called multiple times (as long as the request to eth_getBlockByNumber succeeded the first time)', async () => {
+      await withController(async ({ controller, network }) => {
+        // This mocks eth_getBlockByNumber once by default
+        network.mockEssentialRpcCalls();
+        await withoutCallingGetEIP1559Compatibility({
+          controller,
+          operation: async () => {
+            await controller.initializeProvider();
+          },
         });
+
+        await controller.getEIP1559Compatibility();
+        await controller.getEIP1559Compatibility();
+
+        expect(network.nockScope.isDone()).toBe(true);
       });
     });
   });
-});
+
+  describe('lookupNetwork', () => {
+    describe('if the provider has not been initialized', () => {
+      it('does not update state in any way', async () => {
+        const providerConfig = {
+          type: 'rpc',
+          rpcUrl: 'http://example-custom-rpc.metamask.io',
+          chainId: '0x9999',
+          nickname: 'Test initial state',
+        };
+        const initialState = {
+          provider: providerConfig,
+          networkDetails: {
+            EIPS: {
+              1559: true,
+            },
+          },
+        };
+
+        await withController(
+          {
+            state: initialState,
+          },
+          async ({ controller, network }) => {
+            network.mockEssentialRpcCalls();
+
+            await waitForStateChanges({
+              controller,
+              propertyPath: ['network'],
+              count: 0,
+              operation: async () => {
+                await controller.lookupNetwork();
+              },
+            });
+
+            expect(controller.store.getState()).toStrictEqual({
+              ...initialState,
+              network: 'loading',
+              previousProviderStore: providerConfig,
+            });
```

Comment:
This seems like a pretty confusing way of demonstrating that state doesn't change. At first glance I thought this was showing that a state change _did_ happen, because of the "wait for state changes" helper, and the fact that the expected state was clearly different from the initial state.

But the differences just reflect defaults, and we're in fact waiting for zero state changes.

Would this be simpler?

```suggestion
            const stateAfterConstruction = controller.store.getState();

            await controller.lookupNetwork();

            expect(controller.store.getState()).toStrictEqual(stateAfterConstruction);
```

Or is this meant to attempt detecting additional state changes that might happen after `lookupNetwork` resolves?

Reply:
Yes... you're right that this is confusing. I did want to make sure there were no state changes even after waiting for a bit. `lookupNetwork` doesn't completely await all of its async operations, specifically `_checkInfuraAvailability`. After looking at this again, though, that method doesn't change state, so waiting for no changes to occur shouldn't be necessary.